### PR TITLE
Added the start_component() function

### PR DESCRIPTION
**Pull request by agronholm**
This gives us the component start timeout debugging functionality even in test suites.

### CODE REVIEW

File: src/asphalt/core/_component.py [hunk 1]
```diff
@@ -139,3 +158,155 @@ async def run(self) -> int | None:
...
+) -> None:
+    """
+    Start a component and its subcomponents.
+
+    :param component: the (root) component to start
+    :param timeout: seconds to wait for all the components in the hierarchy to start
+        (default: ``20``; set to ``None`` to disable timeout)
+    :raises RuntimeError: if this function is called without an active :class:`Context`
+    :raises TimeoutError: if the startup of the component hierarchy takes more than
+        ``start_timeout`` seconds
```
Comment (davidbrochart) [inline src/asphalt/core/_component.py:176]:
```suggestion
        ``timeout`` seconds
```

File: src/asphalt/core/_component.py [hunk 2]
```diff
@@ -139,3 +158,155 @@ async def run(self) -> int | None:
...
+            )
+
+        await component.start()
+
+        # Cancel the startup timeout, if any
+        if startup_watcher_scope:
+            startup_watcher_scope.cancel()
+
+    if startup_scope.cancel_called:
+        raise TimeoutError("timeout starting component")
```
Comment (davidbrochart) [inline src/asphalt/core/_component.py:206]: Maybe?
```suggestion
        raise TimeoutError("Timeout starting component")
```

File: tests/test_component.py [hunk 3]
```diff
@@ -168,3 +170,21 @@ async def run(self) -> NoReturn:
...
+
+
+async def test_start_component_timeout() -> None:
+    class StallingComponent(Component):
+        async def start(self) -> None:
+            await sleep(3)
+            pytest.fail("Shouldn't reach this point")
+
+    async with Context():
+        with pytest.raises(TimeoutError, match="timeout starting component"):
```
Comment (davidbrochart) [inline tests/test_component.py:189]:
```suggestion
        with pytest.raises(TimeoutError, match="Timeout starting component"):
```

File: src/asphalt/core/_component.py (hunk 2 above)
Comment (agronholm) [reply src/asphalt/core/_component.py:206]: Existing `raise` statements use messages starting with a lowercase letter though?

Comment (davidbrochart) [reply src/asphalt/core/_component.py:206]: :+1:

**Review by davidbrochart — APPROVED** [review]

**Pull request merged**